app/services/support_interface/ministerial_report_candidates_export.rb

```diff
@@ -0,0 +1,118 @@
+module SupportInterface
+  class MinisterialReportCandidatesExport
+    def self.run_daily
+      data_export = DataExport.create!(
+        name: 'Daily export of the candidates minsiterial report',
+        export_type: :ministerial_report_candidates_export,
+      )
+      DataExporter.perform_async(SupportInterface::MinisterialReportCandidatesExport, data_export.id)
+    end
+
+    def call
+      export_rows = {}
+
+      MinisterialReport::SUBJECTS.each { |subject| export_rows[subject] = column_names }
+
+      export_rows[:split] = column_names
+
+      application_forms.each do |application|
+        subjects = determine_subjects(application)
+
+        states = determine_states(application)
+
+        if candidate_has_no_dominant_subject?(subjects)
+          states.each { |state| export_rows[:split][state] += 1 }
+        else
+          dominant_subject = dominant_subject(subjects)
+          next if dominant_subject.nil?
+
+          states&.each { |state| add_row_values(export_rows, dominant_subject, state) }
+        end
+      end
+
+      export_rows[:total] = export_rows[:primary].merge(export_rows[:secondary]) { |_k, primary_value, secondary_value| primary_value + secondary_value }
+
+      export_rows.map { |subject, value| { subject: subject }.merge!(value) }
+    end
+
+    alias data_for_export call
+
+    def determine_states(application)
```

Comment:
Could this be private too?

Reply:
It could – left it outside so it could be directly tested. Do you think perhaps unnecessary, if covered indirectly by the other tests?

Reply:
to taste. TDD purists will say leaving methods public just to test them is a code smell inviting you to create a separate object to do that work. personally I think this is fine